tests/rptest/tests/cluster_self_config_test.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import re
+
+from ducktape.mark import parametrize, matrix
+
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import CloudStorageType, SISettings
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.utils import LogSearchLocal
+
+
+class ClusterSelfConfigTest(RedpandaTest):
+    def __init__(self, ctx):
+        si_settings = SISettings(
+            ctx,
+            #Force self configuration through setting cloud_storage_url_style to None.
+            cloud_storage_url_style=None,
+            cloud_storage_enable_remote_read=ctx.
+            injected_args["cloud_storage_enable_remote_read"],
+            cloud_storage_enable_remote_write=ctx.
+            injected_args["cloud_storage_enable_remote_write"])
+
+        super(ClusterSelfConfigTest, self).__init__(ctx,
+                                                    si_settings=si_settings)
```

Comment:
nit: since we use python 3, we should use the newer `super().__init__(...) style`